tests/toolchain/install.pm

```diff
@@ -24,23 +24,35 @@ sub run {
     # disable packagekitd
     script_run 'systemctl mask packagekit.service';
     script_run 'systemctl stop packagekit.service';
-    # toolchain channels
-    if (!check_var('ADDONS', 'tcm')) {
-        my $arch = get_var('ARCH');
-        assert_script_run "zypper ar -f http://download.suse.de/ibs/SUSE/Products/SLE-Module-Toolchain/12/$arch/product/ SLE-Module-Toolchain12-Pool";
-        assert_script_run "zypper ar -f http://download.suse.de/ibs/SUSE/Updates/SLE-Module-Toolchain/12/$arch/update/ SLE-Module-Toolchain12-Updates";
+
+    if (is_sle() && !sle_version_at_least('15')) {
+        # toolchain channels
+        if (!check_var('ADDONS', 'tcm')) {
+            my $arch = get_var('ARCH');
+            assert_script_run "zypper ar -f http://download.suse.de/ibs/SUSE/Products/SLE-Module-Toolchain/12/$arch/product/ SLE-Module-Toolchain12-Pool";
+            assert_script_run "zypper ar -f http://download.suse.de/ibs/SUSE/Updates/SLE-Module-Toolchain/12/$arch/update/ SLE-Module-Toolchain12-Updates";
+        }
+        zypper_call('in -t pattern gcc5');
+        zypper_call('up');
+
+        # reboot when runing processes use deleted files after packages update
+        type_string "zypper ps|grep 'PPID' || echo OK | tee /dev/$serialdev\n";
+        if (!wait_serial("OK", 100)) {
+            type_string "shutdown -r now\n";
+            $self->wait_boot;
+            select_console('root-console');
+        }
```

Comment:
hm, I consider the update handling and optional rebooting as something should either apply for SLE15 as well or should not be there at all. So for SLE12 a volatile update repo was used but for SLE15 we use a properly synced proxy-SCC repo per build. IMHO we need to find a better way for SLE12 as well but I am not sure which one that is. I checked on IBS if we can find a build-specific module repo but that is not the way it is done for SLE12 so I give up. Your change here is fine :)